src/DeVinci_frontend/components/SelectModelOption.svelte

```diff
@@ -125,14 +127,24 @@
     };
   };
 
-  const changeModel = (id) => {
-    if ($selectedAiModelId === id) {
```

Comment:
not sure we should remove this check as it could trigger the download for the same model multiple times. Have you tested this?